modules/b2b_logic/records.c

```diff
@@ -117,10 +118,13 @@ b2bl_tuple_t* b2bl_insert_new(struct sip_msg* msg, unsigned int hash_index,
 	int size;
 	str extra_headers={0, 0};
 	str local_contact= server_address;
+	struct sip_uri *ct_uri = NULL;
+	struct to_body *pto;
 
 	if(msg)
 	{
-		if (get_local_contact(msg->rcv.bind_address, NULL, &local_contact) < 0)
+		if (contact_user && msg->to) ct_uri = parse_to_uri(msg);
+		if (get_local_contact(msg->rcv.bind_address, &ct_uri->user, &local_contact) < 0)
```

Comment:
This would crash if the contact user feature is not used and the `ct_uri` pointer is NULL.